lib/cocoapods/generator/embed_frameworks_script.rb

```diff
@@ -159,8 +159,9 @@ def script
         end
         xcframeworks_by_config.each do |config, xcframeworks|
           xcframeworks.select { |xcf| xcf.build_type.dynamic_framework? }.each do |xcframework|
+            target_name = xcframework.target_name
             name = xcframework.name
-            contents_by_config[config] << %(  install_framework "#{Target::BuildSettings::XCFRAMEWORKS_BUILD_DIR_VARIABLE}/#{name}/#{name}.framework"\n)
+            contents_by_config[config] << %(  install_framework "#{Target::BuildSettings::XCFRAMEWORKS_BUILD_DIR_VARIABLE}/#{target_name}/#{name}.framework"\n)
```

Comment:
we already have `config` here, can we do `/#{name}_#{config}/#{name.xcframework}` instead?

Reply:
oh yeah we can also try this instead.

Reply:
actually @amorde we would need to pass the `config` into the `CopyXCFrameworksScript` as well, I guess its still better than adding `target_name` right?

Reply:
oh hm.. do we have that available as an environment variable?

Reply:
`$CONFIGURATION`

Reply:
I will see if that works. I think I tried it but I was still getting some duplicates.

Reply:
@amorde now I remember...using $CONFIGURATION does not work.

Xcode always buils targets despite whether you build for release or debug it just doesnt link them.

This means we will build both SampleLibrary_Debug (actual pod name) and SampleLibrary_Release (actual pod name) however because we use $CONFIGURATION the environment sets it to Debug for both and so the build fails the same way.

```

Showing All Errors Only
Multiple commands produce '/Users/dimitris/Library/Developer/Xcode/DerivedData/Demo_XcFrameworks_Integration-ghazscornxdxcxejwakulleviesc/Build/Products/Debug-iphonesimulator/XCFrameworkIntermediates/SampleLibrary_Debug':
1) That command depends on command in Target 'SampleLibrary_Debug' (project 'Pods'): script phase “[CP] Copy XCFrameworks”
2) That command depends on command in Target 'SampleLibrary_Release' (project 'Pods'): script phase “[CP] Copy XCFrameworks”
```

Reply:
I wonder if I can use target name without polluting the XCFramework class.

Reply:
nope its not easy. The problem is for embed frameworks script for an aggregate target, we do not know which xcframework comes from which target in order to provide the correct path...........

any other suggestions/recommendations?

Reply:
A bit confused. Even if we are always building the target regardless of config doesn't `$CONFIGURATION` have the correct value? Why is `SampleLibrary_Release` modifying `XCFrameworkIntermediates/SampleLibrary_Debug`?

Reply:
`Copy XCFrameworks` should be placing the framework into `XCFrameworkIntermediates/SampleLibrary_$CONFIGURATION/SampleLibrary.framework` right?

Reply:
because the output path of `SampleLibrary_Release` is `XCFrameworkIntermediates/SampleLibrary_$CONFIGURATION` so when I build DEBUG it will be `XCFrameworkIntermediates/SampleLibrary_Debug`. The same applies for `SampleLibrary_Debug` pod. so they both output the same framework.

Reply:
I think the use of the config name was getting me confused. If the pod's name is `SampleLibrary_Release`, then I would expect `Copy XCFrameworks` to place the framework slice into `XCFrameworkIntermediates/SampleLibrary_Release_Release/SampleLibrary_Release.framework` (note the double `_Release`)

Reply:
Synced offline - I understand why this doesn't work now. Both targets have different names but produce the same `SampleLibrary.framework` / `SampleLibrary.xcframework`, and both targets are built regardless of config. The path will be the same unless we incorporate the target's name or some other way to distinguish the two